test/unit/serverless/aws-lambda-stream-response.test.js

```diff
@@ -0,0 +1,1855 @@
+  } catch (err) {
+    plan.equal(err, error)
+    plan.equal(transaction.isActive(), false)
+  }
+  await plan.completed
+
+  function confirmErrorCapture() {
+    const errors = agent.errors.traceAggregator.errors
+    plan.equal(errors.length, 1)
+
+    const noticedError = errors[0]
+    const [, transactionName, message, type] = noticedError
+    plan.equal(transactionName, expectedBgTransactionName)
+    plan.equal(message, errorMessage)
+    plan.equal(type, 'SyntaxError')
+  }
+})
+
+test('should record error event when func is async and error is thrown', async (t) => {
+  const plan = tspl(t, { plan: 10 })
+  const { agent, awsLambda, error, event, responseStream, context } = t.nr
+  agent.on('harvestStarted', function confirmErrorCapture() {
+    const errors = agent.errors.traceAggregator.errors
+    plan.equal(errors.length, 1)
+
+    const noticedError = errors[0]
+    const [, transactionName, message, type] = noticedError
+    plan.equal(transactionName, expectedBgTransactionName)
+    plan.equal(message, errorMessage)
+    plan.equal(type, 'SyntaxError')
+  })
+
+  let transaction
+  const handler = decorateHandler(async (event, responseStream) => {
+    transaction = agent.tracer.getTransaction()
+    const chunks = ['step 1', 'step 2', 'step 3']
+    await writeToResponseStream(chunks, responseStream, 100)
+    responseStream.end()
+    return new Promise(() => {
+      plan.ok(transaction)
+      plan.equal(transaction.type, 'bg')
+      plan.equal(transaction.getFullName(), expectedBgTransactionName)
+      plan.ok(transaction.isActive())
+
+      throw error
+    })
+  })
+  const wrappedHandler = awsLambda.patchLambdaHandler(handler)
+
+  try {
+    await wrappedHandler(event, responseStream, context)
+  } catch (err) {
+    plan.equal(err, error)
+    plan.equal(transaction.isActive(), false)
+  }
+  await plan.completed
+})
+
+test(
+  'should record error event when func is async an UnhandledPromiseRejection is thrown',
+  async (t) => {
+    const plan = tspl(t, { plan: 10 })
+    const { agent, awsLambda, error, event, responseStream, context } = t.nr
+    // this and a few other tests harvest more than once.
+    // since we are using plan based testing im only asserting the first harvest
+    agent.on('harvestStarted', function confirmErrorCapture() {
+      const errors = agent.errors.traceAggregator.errors
+      plan.equal(errors.length, 1)
+
+      const noticedError = errors[0]
+      const [, transactionName, message, type] = noticedError
+      plan.equal(transactionName, expectedBgTransactionName)
+      plan.equal(message, errorMessage)
+      plan.equal(type, 'SyntaxError')
+    })
+
+    let transaction
+    const handler = decorateHandler(async (event, responseStream) => {
+      transaction = agent.tracer.getTransaction()
+      const chunks = ['step 1', 'step 2', 'step 3']
+      await writeToResponseStream(chunks, responseStream, 100)
+      responseStream.end()
```

Comment:
if you update based on my suggestions, ending the stream halts this test so you can't end the stream and then throw a promise rejection